skyportal/handlers/api/candidate/scan_report_item.py

```diff
@@ -1,49 +1,75 @@
-import sqlalchemy as sa
 from astropy.time import Time
-from sqlalchemy import select
 
 from baselayer.app.access import auth_or_token
-from baselayer.app.models import User
 from baselayer.log import make_log
 
-from ....models import Group, Source
+from ....models import Obj, Source
 from ....models.scan_report.scan_report_item import ScanReportItem
 from ....utils.safe_round import safe_round
 from ...base import BaseHandler
 
-log = make_log("api/candidate_scan_report_item")
+log = make_log("api/scan_report_item")
 
 
-def create_scan_report_item(report, obj, saved_infos):
+def create_scan_report_item(session, report, sources_by_obj):
     """
     Parameters
     ----------
+    session: sqlalchemy.orm.Session
     report: skyportal.model.ScanReport
         The scan report to create an item for
-    obj: skyportal.model.Obj
-        The object to create the item for
-    saved_infos: dict
-        The saved infos of the object for each group
+    sources_by_obj: tuple (obj_id, source_ids)
+        The object and link source ids to create the item for
     Returns
     -------
     scan_report_item: skyportal.model.ScanReportItem
     """
-    phot_stats = obj.photstats
-    current_mag = phot_stats[0].last_detected_mag if phot_stats else None
+    obj_id, source_ids = sources_by_obj
+
+    obj = session.scalar(
+        Obj.select(session.user_or_token, mode="read").where(Obj.id == obj_id)
+    )
+    current_mag = obj.photstats[0].last_detected_mag if obj.photstats else None
     current_age = (
-        (Time.now().mjd - phot_stats[0].first_detected_mjd) if phot_stats else None
+        (Time.now().mjd - obj.photstats[0].first_detected_mjd)
+        if obj.photstats
+        else None
     )
 
+    sources = session.scalars(
```

Comment:
you didn't really need to get the source_ids here. You could have just gotten the obj_id earlier, and here query for sources with obj_id and group_id since you already have the groups. Might be easier than using this rather complex json aggregate